File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
""" class rectangle that defines rectangle """


class Rectangle:
    """ Rectangle class """
    def __init__(self, width=0, height=0):
        """ Construct """
        self.height = height
        self.width = width

    @property
    def width(self):
        """ Method """
        return self.__width

    @width.setter
    def width(self, value):
        """ Methode """
        if type(value) is not int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise TypeError("width must be >=0 ")
        self.__width = value

    @property
    def height(self):
        """ Method """
        if type(value) is not int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise TypeError("height must be >= 0")
    
    def area(self):
        """ Return area """
        return self.width * self.height

    def perimeter(self):
        """ Returns perimeter """
        if self.height == 0 or self.width == 0:
            return ""
        p = ""
        for i in range(sellf.height):
            for j in range(self.width):
                p += "#"
        p = p[:-1]
        return p

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
// Write a class Rectangle that defines a rectangle:
// You must use the class notation for defining your class
// The constructor must take 2 arguments width and height
// If w or h is equal to 0 or not a positive integer, create an empty object
module.exports = class Rectangle {
  constructor (width, height) {
    if (width > 0 && height > 0 && width !== undefined && height !== undefined) {
      this.width = width;
      this.height = height;
    }
  }

  // method
  print () {
    for (let x = 0; x < this.height; x++) {
      console.log('X'.repeat(this.width));
    }
  }
};
